src/_macosx.m

```diff
@@ -778,6 +820,23 @@ static CGFloat _get_device_scale(CGContextRef cr)
     Py_RETURN_NONE;
 }
 
+static PyObject*
+FigureManager__get_image_types(PyObject* no_self, PyObject* no_args)
```

Comment:
Do we need this helper function? It looks like above you are only trying to set the pdf icon, so we could even do a simple check for that case explicitly within the `set_icon()` function and return 0 if it isn't supported and just move on.

Reply:
According to the docs, the compatible image formats vary with OS release. I'm under the impression that in practice they're fairly stable. On the other hand, if we know the available formats, we can pick the right image from the install based on what will work. That would beat just trying a bunch of formats and hoping they don't error?

Reply:
But you aren't cycling through the others above? It looks like it is only trying the pdf version right now.

Reply:
I mean, maybe I’m anticipating too much? It’s more pythonic to try and fail than make it complicated. So rather, don’t check if it’s a compatible file type but just set it. If it’s not a valid file, it’ll fail. If we need to support different file types on different versions of macOS (for whatever reason), add the complexity then. Yes?

Reply:
Yep, exactly :) I think that will simplify this PR, and we can add more PRs in the future to expand on capabilities if we find out it is needed.

Reply:
Done